include/nlohmann/detail/input/input_adapters.hpp

```diff
@@ -58,6 +58,16 @@ class file_input_adapter : public input_adapter_protocol
         : m_file(f)
     {}
 
+    file_input_adapter(const file_input_adapter& other) noexcept
+        : m_file(other.m_file)
+    {}
+
+    file_input_adapter& operator=(const file_input_adapter& other) noexcept
+    {
+        m_file = other.m_file;
+        return *this;
+    }
+
```

Comment:
1. Why not just "=default" for both of these?
2. Same comment as above: we should always follow the Rule of 5.

Reply:
Does `-Weffc++` really force defaulting everything?